src/XrdCeph/XrdCephBulkAioRead.cc

```diff
@@ -0,0 +1,187 @@
+#include "XrdCephBulkAioRead.hh"
+
+
+bulkAioRead::bulkAioRead(librados::IoCtx *ct, logfunc_pointer logwrapper, std::string filename, size_t object_sz) {
+  /**
+   * Constructor.
+   *
+   * @param ct                Rados IoContext object
+   * @param logfunc_pointer   Pointer to the function that will be used for logging
+   * @param filename          Name of the file (not ceph object!) that is supposed to be read
+   * @param object_sz         Maximum size of the ceph objects that the file is split into
+   *
+   */
+  context = ct;
+  object_size = object_sz;
+  file_name = filename;
+  log_func = logwrapper;
+}
+
+bulkAioRead::~bulkAioRead() {
+  /**
+   * Destructor. Just clears dynamically allocated memroy.
+   */
+  clear();
+}
+
+void bulkAioRead::clear() {
+  /**
+   * Clear all dynamically alocated memory
+   */
+  buffers.clear();
+  operations.clear();
```

Comment:
If `~CmplPtr()` can be made to ensure that all operations have been terminated, it might be better to clear `operations` before `buffers`, which contains things that the operations expect to exist.

Reply:
Done.